src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java

```diff
@@ -1,11 +1,215 @@
+import org.apache.brooklyn.util.core.task.system.ProcessTaskWrapper;
+import org.apache.brooklyn.util.guava.Maybe;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.brooklyn.util.time.Duration;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 public class TerraformConfigurationImpl extends SoftwareProcessImpl implements TerraformConfiguration {
 
+    private static final Duration FEED_UPDATE_PERIOD = Duration.seconds(30);
+
+    private SshFeed sshFeed;
+
+    private FunctionFeed functionFeed;
+
+    private final AtomicBoolean configurationChangeInProgress = new AtomicBoolean(false);
+
+    private final AtomicBoolean configurationIsApplied = new AtomicBoolean(false);
+
+    private final Map<String, Object> lastCommandOutputs = Collections.synchronizedMap(Maps.<String, Object> newHashMapWithExpectedSize(3));
+
+    @Override
+    public void init() {
+        super.init();
+        checkConfiguration();
+    }
+
+    @Override
+    protected void connectSensors() {
+        super.connectSensors();
+        connectServiceUpIsRunning();
+
+        addFeed(functionFeed = FunctionFeed.builder()
+            .entity(this)
+            .period(FEED_UPDATE_PERIOD)
+            .poll(new FunctionPollConfig<Object, Boolean>(CONFIGURATION_IS_APPLIED)
+                .callable(new Callable<Boolean>() {
+                    @Override
+                    public Boolean call() throws Exception {
+                        return configurationIsApplied.get();
+                    }
+                }))
+            .build());
+
+        Maybe<SshMachineLocation> machine = Locations.findUniqueSshMachineLocation(getLocations());
+        if (machine.isPresent()) {
+            addFeed(sshFeed = SshFeed.builder()
+                .entity(this)
+                .period(FEED_UPDATE_PERIOD)
+                .machine(machine.get())
+                .poll(new SshPollConfig<String>(SHOW)
+                    .command(getDriver().makeTerraformCommand("show -no-color"))
+                    .onSuccess(new Function<SshPollValue, String>() {
+                        @Override
+                        public String apply(SshPollValue input) {
+                            String output = input.getStdout();
+                            if (Strings.isBlank(output)) output = "No configuration is applied.";
+                            lastCommandOutputs.put(SHOW.getName(), output);
+
+                            return output;
+                        }})
+                    .onFailure(new Function<SshPollValue, String>() {
+                        @Override
+                        public String apply(SshPollValue input) {
+                            if (configurationChangeInProgress.get() && lastCommandOutputs.containsKey(SHOW.getName()))
+                                return (String) lastCommandOutputs.get(SHOW.getName());
```

Comment:
Can return [`FeedConfig.UNCHANGED`](https://github.com/apache/incubator-brooklyn/blob/master/core/src/main/java/org/apache/brooklyn/core/feed/AttributePollHandler.java#L218) no need to use `lastCommandOutputs`.